tests/chaos/conftest.py

```diff
@@ -186,25 +187,34 @@ def rebooting_master_node(
 
 @pytest.fixture()
 def pod_deleting_process(request, admin_client):
+    """
+    A pytest fixture to create and manage a pod-deleting process during a test.
+    This fixture sets up a process to delete pods matching a specific prefix in a given namespace
+    with specified parameters. The process starts before the test begins and ensures proper
+    cleanup after the test ends.
+    Args:
+        request (SubRequest): Pytest's request object that provides access to the parameters passed to the fixture.
+        admin_client (DynamicClient): The Kubernetes dynamic client used to interact with the cluster.
+    Returns:
+        PodDeletingProcess: A dataclass instance containing the process object, the namespace name,
+        and the pod prefix, allowing easy access to these details in the test.
+    """
+    PodDeletingProcess = namedtuple("PodDeletingProcess", ["process", "namespace_name", "pod_prefix"])
+
     pod_prefix = request.param["pod_prefix"]
     namespace_name = request.param["namespace_name"]
-    pod_deleting_process = create_pod_deleting_process(
+    process = create_pod_deleting_process(
         dyn_client=admin_client,
         pod_prefix=pod_prefix,
         namespace_name=namespace_name,
         ratio=request.param["ratio"],
         interval=request.param["interval"],
         max_duration=request.param["max_duration"],
     )
-    pod_deleting_process.start()
-    yield pod_deleting_process
-    terminate_process(process=pod_deleting_process)
-
-    pod_deleting_process_recover(
-        resource=request.param["resource"],
-        namespace=namespace_name,
-        pod_prefix=pod_prefix,
-    )
+    process.start()
+    # Yield a wrapper object that includes the process and its related information
+    yield PodDeletingProcess(process=process, namespace_name=namespace_name, pod_prefix=pod_prefix)
```

Comment:
May be return the regular dict instead of namedtuple?

Reply:
+1

Reply:
Done